packages/medusa/src/services/cart.ts

```diff
@@ -1641,11 +1641,20 @@ class CartService extends TransactionBaseService {
           })
         )
 
-        cart.discounts.forEach((discount) => {
-          if (discountsMap.has(discount.id)) {
-            discountsMap.delete(discount.id)
-          }
-        })
+        await promiseAll(
+          cart.discounts.map(async (discount) => {
+            try {
+              await this.discountService_
+                .withTransaction(transactionManager)
+                .validateDiscountForCartOrThrow(cart, discount)
```

Comment:
todo: Remove this is already above.
https://github.com/medusajs/medusa/blob/884428a1b573e499d7659aefed639bf797147428/packages/medusa/src/services/cart.ts#L1632-L1634